src/Illuminate/Contracts/Database/Eloquent/Factory.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Illuminate\Contracts\Database\Eloquent;
+
+use ArrayAccess;
+use Faker\Generator as Faker;
+
+interface Factory extends ArrayAccess
```

Comment:
I don't think we need to extend ArrayAccess here. It would be better to just keep the interface at its sole purpose. The fact that the actual implementation is an implementation of `ArrayAccess` is a separate matter and not a requirement for a factory.

Reply:
I reverted the change.  ArrayAccess is now on implementation.